text/nodejs/0015-dependencies.md

```diff
@@ -0,0 +1,29 @@
+# Decide which `nodejs` dependencies will be Paketo-hosted
+
+## Proposal
+
+The following dependencies should be kept as Paketo-hosted dependencies:
+* [node-engine](https://github.com/paketo-buildpacks/node-engine/blob/main/buildpack.toml)
+
+The following dependencies should be removed as Paketo-hosted dependencies:
+* [yarn](https://github.com/paketo-buildpacks/yarn/blob/main/buildpack.toml)
+
+## Rationale
+
+### node-engine
+
+Keep this as a Paketo-hosted dependency.
+
+The Paketo node-engine buildpack supports using the system's CA store, which is not a default feature in node. To enable this feature, node
+must be compiled from source with the `--openssl-use-def-ca-store` flag. Therefore, this RFC does not propose removing node as a
+Paketo-hosted dependency.
```

Comment:
I'd like to see a deeper investigation on this point. I was able to use the Node dependency provided by the Node.js project to confirm the CA certs workflows with 2 changes to the `node-engine` buildpack:
1. use the node tarball from upstream and use `strip-components = 1` in the `buildpack.toml` to make sure the tarball get extracted correctly into the layer
2. set `NODE_OPTIONS=--use-openssl-ca` as a default env var on the node layer

Reply:
I'll take a look